consensus/src/quorum_store/batch_store.rs

```diff
@@ -64,28 +64,38 @@ enum StorageMode {
 struct QuotaManager {
     memory_balance: usize,
     db_balance: usize,
+    batch_balance: usize,
     memory_quota: usize,
     db_quota: usize,
+    batch_quota: usize,
 }
 
 impl QuotaManager {
-    fn new(max_db_balance: usize, max_memory_balance: usize) -> Self {
-        assert!(max_db_balance >= max_memory_balance);
+    fn new(db_quota: usize, memory_quota: usize, batch_quota: usize) -> Self {
+        assert!(db_quota >= memory_quota);
         Self {
             memory_balance: 0,
             db_balance: 0,
-            memory_quota: max_memory_balance,
-            db_quota: max_db_balance,
+            batch_balance: 0,
+            memory_quota,
+            db_quota,
+            batch_quota,
         }
     }
 
     pub(crate) fn update_quota(&mut self, num_bytes: usize) -> anyhow::Result<StorageMode> {
+        if self.batch_balance + 1 > self.batch_quota {
+            bail!("Batch quota exceeded ");
```

Comment:
Add a counter for EXCEEDED_BATCH_QUOTA_COUNT similar to EXCEEDED_STORAGE_QUOTA_COUNT